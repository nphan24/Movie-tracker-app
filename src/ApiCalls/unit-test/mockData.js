export const mockMovieData = {
  results: [
    {
      original_language: 'en',
      original_title: 'Jumanji: Welcome to the Jungle',
      overview:
        "The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
      popularity: 101.350698,
      poster_path: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
      release_date: '2017-12-09',
      title: 'Jumanji: Welcome to the Jungle',
      video: false,
      vote_average: 6.5,
      vote_count: 3115
    }
  ]
};

export const cleanedMovie = [
  {
    date: 'Release Date: 2017-12-09',
    id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];

export const movieInfo = {
  movie_id: '1',
  user_id: '24',
  title: 'hello',
  poster_path: '/hello',
  release_date: '04-01-18',
  vote_average: '-5',
  overview: 'Its not real'
};

export const mockUser = {
  id: '5',
  name: 'r',
  email: 'r@r.r',
  password: 'r'
};

export const mockFavorite = {
  date: 'Release Date: 2017-12-09',
  movie_id: 101.350698,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview:
    "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
  title: 'Jumanji: Welcome to the Jungle',
  vote: 'Rating: 6.5'
};

export const mockRemainingFavorite = {
  date: 'Release Date: 2018-10-5',
  movie_id: 10,
  image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
  overview: 'Summary: The tables are turned as four teenagers.',
  title: 'Fantastic Fours',
  vote: 'Rating: 2.5'
};

export const mockFavoritesArray = [
  {
    date: 'Release Date: 2018-10-5',
    movie_id: 10,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview: 'Summary: The tables are turned as four teenagers.',
    title: 'Fantastic Fours',
    vote: 'Rating: 2.5'
  },
  {
    date: 'Release Date: 2017-12-09',
    movie_id: 101.350698,
    image: '/bXrZ5iHBEjH7WMidbUDQ0U2xbmr.jpg',
    overview:
      "Summary: The tables are turned as four teenagers are sucked into Jumanji's world - pitted against rhinos, black mambas and an endless variety of jungle traps and puzzles. To survive, they'll play as characters from the game.",
    title: 'Jumanji: Welcome to the Jungle',
    vote: 'Rating: 6.5'
  }
];
