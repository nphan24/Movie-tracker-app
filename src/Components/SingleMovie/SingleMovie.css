.single-card {
  position: fixed;
  z-index: 20;
  top: 0;
  color: #fff;
  background-color: rgba(0, 0, 0, 0.8);
  height: 100%;
  width: 100%;
}

.movie-poster {
  height: 600px;
}

.overview {
  width: 500px;
  margin: auto;
}

.close {
  position: absolute;
  top: 20px;
  right: 30%;
  color: white;
  font-size: 20px;
  text-decoration: none;
}
