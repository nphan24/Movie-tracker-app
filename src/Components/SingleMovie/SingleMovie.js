import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from "react-redux";
import { postToFavorites } from "../../ApiCalls/postToFavorites";
import * as Actions from "../../Actions/";
import { deleteFromFavorites } from "../../ApiCalls/deleteFromFavorites";

export const SingleMovie = (props) => {
  const { title, overview, release_date, vote_average, poster_path } = props.movie;

  const toggleFavorite = async (movie) => {
    if (!props.user.name) {
      const error = "Please Sign In to Add Favorites";
      props.promptSignin(error);
    } else if (!props.favorites.find(fav => fav.movie_id === movie.movie_id)) {
      await postToFavorites({ ...movie, user_id: props.user.id });
      props.addFavorite( movie );
    } else {
      props.removeFavorite(movie.movie_id);
      deleteFromFavorites(movie.movie_id, props.user.id); 
    }
  };

  return (
      <article className="single-card">
        <h1>{title}</h1>
        <img src={`https://image.tmdb.org/t/p/w500/${poster_path}`}/>
        <h3>{release_date}</h3>
        <p>{vote_average}</p>
        <p>{overview}</p>
        <Link to={`/${props.lastPath}`}>
          <button>Back</button>
        </Link>
      </article>
  )
}

export const mapStateToProps = state => ({
  user: state.user,
  favorites: state.favorites
});

export const mapDispatchToProps = dispatch => ({
  addFavorite: movie => dispatch(Actions.addFavorite(movie)),
  removeFavorite: id => dispatch(Actions.removeFavorite(id)),
  promptSignin: error => dispatch(Actions.promptSignIn(error))
});

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(SingleMovie));