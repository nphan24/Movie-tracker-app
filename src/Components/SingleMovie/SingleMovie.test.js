import React from 'react';
import { shallow } from 'enzyme';
import SingleMovie from './SingleMovie';
import * as mock from '../../ApiCalls/unit-test/mockData.js';

describe('SingleMovie', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<SingleMovie movie={mock.movieInfo}/>);
    expect(wrapper).toMatchSnapshot();
  });
});