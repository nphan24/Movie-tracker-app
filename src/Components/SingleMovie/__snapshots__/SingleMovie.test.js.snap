// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleMovie should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleMovie
    movie={
        Object {
            "movie_id": "1",
            "overview": "Its not real",
            "poster_path": "/hello",
            "release_date": "04-01-18",
            "title": "hello",
            "user_id": "24",
            "vote_average": "-5",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="single-card-title"
>
          hello
</h1>,
        <div
          className="poster-container"
>
          <img
                    alt="movie poster"
                    className="movie-poster"
                    src="https://image.tmdb.org/t/p/w500//hello"
          />
          <div
                    className="single-card-info"
          >
                    <p
                              className="overview"
                    >
                              Its not real
                    </p>
                    <h3
                              className="single-card-date"
                    >
                              04-01-18
                    </h3>
                    <p
                              className="single-card-vote"
                    >
                              -5
                    </p>
          </div>
</div>,
        <Link
          className="close"
          replace={false}
          to="/undefined"
>
          x
</Link>,
      ],
      "className": "single-card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "hello",
          "className": "single-card-title",
        },
        "ref": null,
        "rendered": "hello",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="movie poster"
              className="movie-poster"
              src="https://image.tmdb.org/t/p/w500//hello"
/>,
            <div
              className="single-card-info"
>
              <p
                            className="overview"
              >
                            Its not real
              </p>
              <h3
                            className="single-card-date"
              >
                            04-01-18
              </h3>
              <p
                            className="single-card-vote"
              >
                            -5
              </p>
</div>,
          ],
          "className": "poster-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "movie poster",
              "className": "movie-poster",
              "src": "https://image.tmdb.org/t/p/w500//hello",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="overview"
>
                  Its not real
</p>,
                <h3
                  className="single-card-date"
>
                  04-01-18
</h3>,
                <p
                  className="single-card-vote"
>
                  -5
</p>,
              ],
              "className": "single-card-info",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Its not real",
                  "className": "overview",
                },
                "ref": null,
                "rendered": "Its not real",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "04-01-18",
                  "className": "single-card-date",
                },
                "ref": null,
                "rendered": "04-01-18",
                "type": "h3",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "-5",
                  "className": "single-card-vote",
                },
                "ref": null,
                "rendered": "-5",
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "x",
          "className": "close",
          "replace": false,
          "to": "/undefined",
        },
        "ref": null,
        "rendered": "x",
        "type": [Function],
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="single-card-title"
>
            hello
</h1>,
          <div
            className="poster-container"
>
            <img
                        alt="movie poster"
                        className="movie-poster"
                        src="https://image.tmdb.org/t/p/w500//hello"
            />
            <div
                        className="single-card-info"
            >
                        <p
                                    className="overview"
                        >
                                    Its not real
                        </p>
                        <h3
                                    className="single-card-date"
                        >
                                    04-01-18
                        </h3>
                        <p
                                    className="single-card-vote"
                        >
                                    -5
                        </p>
            </div>
</div>,
          <Link
            className="close"
            replace={false}
            to="/undefined"
>
            x
</Link>,
        ],
        "className": "single-card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "hello",
            "className": "single-card-title",
          },
          "ref": null,
          "rendered": "hello",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="movie poster"
                className="movie-poster"
                src="https://image.tmdb.org/t/p/w500//hello"
/>,
              <div
                className="single-card-info"
>
                <p
                                className="overview"
                >
                                Its not real
                </p>
                <h3
                                className="single-card-date"
                >
                                04-01-18
                </h3>
                <p
                                className="single-card-vote"
                >
                                -5
                </p>
</div>,
            ],
            "className": "poster-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "movie poster",
                "className": "movie-poster",
                "src": "https://image.tmdb.org/t/p/w500//hello",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="overview"
>
                    Its not real
</p>,
                  <h3
                    className="single-card-date"
>
                    04-01-18
</h3>,
                  <p
                    className="single-card-vote"
>
                    -5
</p>,
                ],
                "className": "single-card-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Its not real",
                    "className": "overview",
                  },
                  "ref": null,
                  "rendered": "Its not real",
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "04-01-18",
                    "className": "single-card-date",
                  },
                  "ref": null,
                  "rendered": "04-01-18",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "-5",
                    "className": "single-card-vote",
                  },
                  "ref": null,
                  "rendered": "-5",
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "x",
            "className": "close",
            "replace": false,
            "to": "/undefined",
          },
          "ref": null,
          "rendered": "x",
          "type": [Function],
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
